File: frontend/src/types/session.ts
export interface Session {
  id: string
  name: string
  source_type: string
  source_filename: string | null
  status: string
  created_at: number
  closed_at: number | null
  total_frames: number
  final_state: string | null
}

export type SourceType = 'WEBCAM' | 'VIDEO' | 'IMAGE'

File: backend/app/models/schemas/session.py
from pydantic import BaseModel
from typing import Optional


class SessionCreate(BaseModel):
    name: str = ""
    source_type: str  # IMAGE | VIDEO | WEBCAM


class SessionOut(BaseModel):
    id: str
    name: str
    source_type: str
    source_filename: Optional[str]
    status: str
    created_at: int
    closed_at: Optional[int]
    total_frames: int
    final_state: Optional[str]

    model_config = {"from_attributes": True}
